hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/impl/OperationExecutorImpl.java

```diff
@@ -546,6 +548,43 @@ private static void awaitTermination(OperationThread[] operationThreads) {
         }
     }
 
+    @Override
+    public void flush() throws InterruptedException {
+        // Basically, what we do here is just blocking all the threads by
+        // submitting a blocking barrier task on them. When all the threads are
+        // blocked, we may safely assume that all the tasks, submitted before
+        // the barriers, are completed.
+
+        final CountDownLatch latch = new CountDownLatch(genericThreads.length + partitionThreads.length);
+        final AtomicReference<InterruptedException> seenInterruptedException = new AtomicReference<InterruptedException>(null);
+        Runnable barrier = new Runnable() {
+            @Override
+            public void run() {
+                latch.countDown();
+                try {
+                    latch.await();
+                } catch (InterruptedException e) {
+                    seenInterruptedException.set(e);
+                }
+            }
+        };
+
+        for (int i = 0; i < genericThreads.length; ++i) {
+            genericQueue.add(barrier, false);
+            genericQueue.add(barrier, true);
+        }
+        for (PartitionOperationThread partitionThread : partitionThreads) {
+            partitionThread.queue.add(barrier, false);
+            partitionThread.queue.add(barrier, true);
+        }
+
+        if (seenInterruptedException.get() != null) {
```

Comment:
You can't guarantee that you will see the interrupt here, can you?
All we know at this stage is that the `barrier` task is in the queue, (i.e pending execution). I guess the only guaranteed approach is with an await_timeout and re-check approach. WDYT?